### ISSUE TITLE
If kube/config file contains 'localhost', do not change it to the local IP address

### ISSUE THREAD

**Issue by erniedavisAA** [rancher-sandbox/rancher-desktop#1503]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://github.com/rancher-sandbox/rancher-desktop/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

On every start of Rancher Desktop, it modifies the kubeconfig file to update the IP address for the local machine. 

### Proposed Solution

Instead, check the existing value for 'server:' and if it contains 'localhost' (which means it was overridden by the user), then do not update it.

### Additional Information

The localhost entry is necessary to support context switching from VPN to Standalone networks via use of wsl-vpnkit.

**Comment by ericpromislow** [comment]
Thanks, sounds reasonable.

**Comment by jandubois** [comment]
From another Slack thread:

> I've got a strange issue with my kubeconfig ~/.kube/config. Besides the context "rancher-desktop" I have also other contexts configured e.g. gcloud (GKE). When I reset my rancher-desktop kubernetes environment my config attribute "proxy-url: http://localhost:8888/" from a foreign gke context config gets removed. It seems as the rancher-desktop logic somehow modifies the other contexts as well.

Rancher Desktop should not modify **any** settings besides the ones related to the `rancher-desktop` context!

**Comment by spicefather** [comment]
Adding on to this, I've run into issues with Rancher modifying user certificate/key paths. My company's k8s environment requires them to be relative to work properly, but Rancher is making them absolute. It's easy to fix (once I figured out the problem) but I 100% agree Rancher should never modify configurations it doesn't own.

**Comment by jandubois** [comment]
> Adding on to this, I've run into issues with Rancher modifying user certificate/key paths. My company's k8s environment requires them to be relative to work properly, but Rancher is making them absolute.

This happens because we read the kubeconfig using the `loadFromFile` function from the https://github.com/kubernetes-client/javascript library: https://github.com/rancher-sandbox/rancher-desktop/blob/206c716dd55e3a768b3398f54eb10f1163521759/src/k8s-engine/k3sHelper.ts#L835

```ts
    public loadFromFile(file: string, opts?: Partial<ConfigOptions>): void {
        const rootDirectory = path.dirname(file);
        this.loadFromString(fs.readFileSync(file, 'utf8'), opts);
        this.makePathsAbsolute(rootDirectory);
    }
```

This function could easily be inlined without the `makePathsAbsolute(rootDirectory)` call at the end.

**Comment by jandubois** [comment]
@spicefather Pathnames should no longer be made absolute once #2590 is merged. The `proxy-url` setting should also be preserved then.

Not replacing the local cluster definition whenever Rancher Desktop starts is a separate issue, so we'll keep this Github issue open to track it. I'll change this to an enhancement request though, as the current behaviour is by design.
